qi-lib/flow/impl.rkt

```diff
@@ -171,19 +167,11 @@
       (call-with-values (λ () (apply b args)) list)))
   (apply values (apply append results)))
 
-(define (->boolean v)
-  (not (not v)))
-
-(define true.
-  (procedure-rename (const #t)
-                    'true.))
-
-(define false.
-  (procedure-rename (const #f)
-                    'false.))
-
-(define exists ormap)
+(define (->boolean v) (and v #t))
+(define true.  (thunk* #t))
+(define false. (thunk* #f))
```

Comment:
What's the benefit to using `thunk*` over `const`? Are these faster? Also, I assume `procedure-rename` is there to provide helpful error messages. Is that not needed here?

Reply:
Yes, using thunk* doesn't require renaming, just like lambda.